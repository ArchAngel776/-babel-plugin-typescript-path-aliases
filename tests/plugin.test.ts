import { resolve } from "path"
import { transformFileSync } from "@babel/core"
import plugin from "../src/index"


const sourceCodeFile = resolve(__dirname, "code.ts")

test("Test", () =>
{
    const result = transformFileSync(sourceCodeFile, { plugins: [ "@babel/plugin-syntax-typescript", plugin ] })
    const output = result?.code

    expect(output).toBeTruthy()
    expect(output).toMatchSnapshot()
})