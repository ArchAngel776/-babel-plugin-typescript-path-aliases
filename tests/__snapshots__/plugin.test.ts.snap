// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
""use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = printPluginCode;
var _index = _interopRequireDefault(require("../src/index"));
require("../src/interfaces/AliasesData");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function printPluginCode() {
  console.log(_index.default.toString());
}"
`;
